contrib/fetch-offline-data/instana/fetch_instana_app_metrics.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import os
+import time
+
+import requests
+import json
+import argparse
+from datetime import datetime, timedelta
+
+import urllib3
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+def fetch_instana_application_catalog(url, token):
+    instana_api_url = f"{url}/api/application-monitoring/catalog/metrics"
+
+    headers = {
+        "Authorization": f"apiToken {token}",
+        "Content-Type": "application/json"
+    }
+
+    response = requests.get(instana_api_url, headers=headers, verify=False)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        logging.error("Error fetching plugins:", response.text)
+        return None
+
+def fetch_instana_application_metrics(url, token, metric_ids, start_time, end_time, granularity=1):
+    instana_api_url = f"{url}/api/application-monitoring/metrics/applications"
+
+    s_time = int(start_time.timestamp()) * 1000
+    f_time = int(end_time.timestamp()) * 1000
+    result = []
+    while s_time < f_time:
+        e_time = s_time + granularity * 600 * 1000
+        logging.info(f"Getting metric from {s_time} to {e_time}")
+        params = {
+            "start": s_time,  # Convert to milliseconds
+            "end": e_time,  # Convert to milliseconds
+        }
+
+        headers = {
+            "Authorization": f"apiToken {token}",
+            "Content-Type": "application/json"
+        }
+        metrics = [{"metric" : i, "aggregation":"PER_SECOND", "granularity": granularity} for i in metric_ids]
+        body = {
+            "metrics": metrics,
+            "order":{"by": "timestamp", "direction": "DESC"},
```

Comment:
Is `DESC` order what we expect? I think the default for prometheus provided data is increasing in timestamp.